lib/grandpa/tracker_votes.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"container/list"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto/ed25519"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// votesTracker tracks vote messages that could
+// not be processed, and removes the oldest ones once
+// its maximum capacity is reached.
+// It is NOT THREAD SAFE to use.
+type votesTracker struct {
+	// map of vote block hash to authority ID (ed25519 public Key)
+	// to data (peer id + message + tracking linked list element pointer)
+	mapping map[common.Hash]authorityIDToData
+	// double linked list of block hash + authority ID
+	// to track the order vote messages were added in.
+	linkedList *list.List
```

Comment:
nit: maybe a more specific name

Reply:
Any suggestion in mind?

Mine would be `linkedListOfBlockHashAndAuthorityID` hahaha but that feels like overkill-super-mega-long name.